packages/kit-headless/src/components/pagination/pagination.tsx

```diff
@@ -1,231 +1,135 @@
- * V normalStyles: The styles of the inactive page buttons
- * V customArrowTexts: { previous, next }
- * disabled: enable/disable paginator
- * color: primary | secondary
- * variant: outlined (show border without bg)
- * shape: rounded | square
- * size: 'sm' | 'md' | 'lg'
- * paginationRange (see https://mui.com/material-ui/react-pagination/#pagination-ranges)
- *
- */
-export interface PaginationProps extends PaginationOptions {
-  pages: number;
-  page: number;
-  onPaging$: PropFunction<(index: number) => void>;
-  RenderItem?: Component<PaginationButtonProps>;
-  RenderDivider?: Component<object>;
-}
+type ArrowLabels = {
+  previous: string;
+  next: string;
+};
 
-export interface PaginationOptions {
-  boundaryCount?: number;
+export interface PaginationProps {
+  selectedPage: number;
+  totalPages: number;
+  onPageChange$: PropFunction<(page: number) => void>;
+
+  // configuration
   siblingCount?: number;
+  boundaryCount?: number;
   hidePrevButton?: boolean;
   hideNextButton?: boolean;
-  showFirstButton?: boolean;
-  showLastButton?: boolean;
-  activeClass?: string;
-  defaultClass?: string;
-  labels?: PaginationButtonLabels;
-}
-
-export interface PaginationButtonProps {
-  onClick$: PropFunction<() => void>;
   disabled?: boolean;
-  'aria-label': string;
-  'aria-current'?: boolean;
-  key?: string | number;
-  activeClass?: string;
-  defaultClass?: string;
-  value: PaginationButtonValue;
-  labels?: PaginationButtonLabels;
-}
+  customArrowTexts?: ArrowLabels;
 
-export type PaginationButtonValue =
-  | 'prev'
-  | 'next'
-  | 'first'
-  | 'last'
-  // | 'divider'
-  // | 'start-ellipsis'
-  // | 'end-ellipsis'
-  | number
-  | string;
-
-export interface PaginationButtonLabels {
-  first?: string;
-  last?: string;
-  next?: string;
-  prev?: string;
+  // styling
+  class?: string;
+  gap?: string;
+  defaultClass?: string;
+  selectedClass?: string;
+  dividerClass?: string;
 }
 
-const range = (start: number, end: number) => {
-  const length = end - start + 1;
-  return Array.from({ length }, (_, i) => start + i);
-};
-
-export const getPaginationButtons = (
-  page: number,
-  count: number,
-  labels: PaginationButtonLabels | undefined,
-  {
-    boundaryCount = 1,
-    siblingCount = 1,
-    hidePrevButton,
-    hideNextButton,
-    showFirstButton,
-    showLastButton,
-  }: PaginationOptions,
-): PaginationButtonValue[] => {
-  const startPages = range(1, Math.min(boundaryCount, count));
-  const endPages = range(Math.max(count - boundaryCount + 1, boundaryCount + 1), count);
-
-  const siblingsStart = Math.max(
-    Math.min(
-      page - siblingCount, // Natural start
-      count - boundaryCount - siblingCount * 2 - 1, // Lower boundary when page is high
-    ),
-    boundaryCount + 2, // Greater than startPages
+export const Pagination = component$<PaginationProps>((props) => {
+  const {
+    selectedPage,
+    totalPages,
+    onPageChange$,
+    // configuration
+    siblingCount,
```

Comment:
Hi Jack, thank you for the review.
I'm not sure about this comment:
do you mean I should add some comments to better describe siblingCount (and boundaryCount)?
In this case I may add them directly to the type, above each property but they are explained in the documentation so it may be redundant.

But I want explain you the reason I have used these comments above some proporties. I mean:
`// configuration`
and
`// styling`

I wrote these comments to split  the properties in two sub-groups: all the props for styling such as (gap, defaultClass, selectedClass and so on) and all about configuration.

But I have a suggestion:
I may split tthe type  in multiiple types and use intersection. I mean:

```ts
export type PaginationStyling = {
  class?: string;
  gap?: string;
  defaultClass?: string;
  selectedClass?: string;
  dividerClass?: string;
}

export type PaginationConfig = {
siblingCount?: number;
  boundaryCount?: number;
  hidePrevButton?: boolean;
  hideNextButton?: boolean;
  disabled?: boolean;
  customArrowTexts?: ArrowLabels;
}
```
and:

```ts
component$<PaginationProps & PaginationConfig & PaginationStyle>
```

What do you think?

Reply:
Update: I have removed the comments and split the Pagination type in several alias types.